osm2lanes/src/transform/tags_to_lanes/modes/non_motorized.rs

```diff
@@ -1,133 +1,38 @@
-        Self::Travel {
-            direction: None,
-            designated,
-            width: Some(locale.travel_width(&designated, HighwayType::Footway)),
-            max_speed: None,
-            access: None,
-        }
-    }
-    fn forward(designated: Designated, locale: &Locale) -> Self {
-        Self::Travel {
-            direction: Some(Direction::Forward),
-            designated,
-            width: Some(locale.travel_width(&designated, HighwayType::Cycleway)),
-            max_speed: None,
-            access: None,
-        }
-    }
-
-    fn backward(designated: Designated, locale: &Locale) -> Self {
-        Self::Travel {
-            direction: Some(Direction::Backward),
-            designated,
-            width: Some(locale.travel_width(&designated, HighwayType::Cycleway)),
-            max_speed: None,
-            access: None,
-        }
-    }
-}
-
-#[allow(clippy::unnecessary_wraps)]
+#[allow(clippy::unnecessary_wraps, clippy::restriction)]
 pub(in crate::transform::tags_to_lanes) fn non_motorized(
     tags: &Tags,
-    locale: &Locale,
-    road: &RoadBuilder,
-) -> Result<Option<RoadFromTags>, RoadError> {
+    _locale: &Locale,
+    road: &mut RoadBuilder,
+    warnings: &mut RoadWarnings,
+) -> Result<(), TagsToLanesMsg> {
     if road.highway.is_supported_non_motorized() {
         log::trace!("non-motorized");
     } else {
         log::trace!("motorized");
-        return Ok(None);
+        return Ok(());
     }
     // Easy special cases first.
-    if tags.is(HIGHWAY, "steps") || tags.is(HIGHWAY, "path") {
-        return Ok(Some(RoadFromTags {
-            road: Road {
-                lanes: vec![Lane::foot(locale)],
-                highway: road.highway.clone(),
-            },
-            warnings: RoadWarnings::new(if tags.is(HIGHWAY, "steps") {
-                vec![TagsToLanesMsg::unimplemented(
-                    "steps becomes sidewalk",
-                    tags.subset(&[HIGHWAY]),
-                )]
-            } else {
-                Vec::new()
-            }),
-        }));
-    }
-
-    // Eventually, we should have some kind of special LaneType for shared walking/cycling paths of
-    // different kinds. Until then, model by making bike lanes and a shoulder for walking.
-
-    // If it just allows foot traffic, simply make it a sidewalk. For most of the above highway
-    // types, assume bikes are allowed, except for footways, where they must be explicitly
-    // allowed.
-    if tags.is("bicycle", "no")
-        || (tags.is(HIGHWAY, "footway") && !tags.is_any("bicycle", &["designated", "yes"]))
-    {
-        return Ok(Some(RoadFromTags {
-            road: Road {
-                lanes: vec![Lane::foot(locale)],
-                highway: road.highway.clone(),
-            },
-            warnings: RoadWarnings::default(),
-        }));
-    }
-    // Otherwise, there'll always be a bike lane.
-
-    let mut forward_side = vec![Lane::forward(Designated::Bicycle, locale)];
```

Comment:
I think this was essentially dead code -- if the road allowed a mix of walking and cycling, another part already would've filled out cycle lanes